ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/MigrationConfig.java

```diff
@@ -0,0 +1,139 @@
+          PropertiesUtil.loadProperties(new File("ksql-migrations.properties"));
+      return new MigrationConfig(configsMap, getServiceId(configsMap));
+    } catch (KsqlException e) {
+      LOGGER.error(e.getMessage());
+      System.exit(1);
+    }
+    return null;
+  }
+
+  private MigrationConfig(final Map<String, String> configs, final String id) {
+    super(new ConfigDef()
+        .define(
+            KSQL_SERVER_URL,
+            Type.STRING,
+            "",
+            Importance.HIGH,
+            "The URL for the KSQL server"
+        )
+        .define(
+            KSQL_MIGRATIONS_STREAM_NAME,
+            Type.STRING,
+            KSQL_MIGRATIONS_STREAM_NAME_DEFAULT,
+            Importance.MEDIUM,
+            "The name of the migration stream. It defaults to "
+                + KSQL_MIGRATIONS_STREAM_NAME_DEFAULT
+        ).define(
+            KSQL_MIGRATIONS_TABLE_NAME,
+            Type.STRING,
+            KSQL_MIGRATIONS_TABLE_NAME_DEFAULT,
+            Importance.MEDIUM,
+            "The name of the migration table. It defaults to "
+                + KSQL_MIGRATIONS_TABLE_NAME_DEFAULT
+        ).define(
+            KSQL_MIGRATIONS_STREAM_TOPIC_NAME,
+            Type.STRING,
+            id + "ksql_" + configs.getOrDefault(KSQL_MIGRATIONS_STREAM_NAME, KSQL_MIGRATIONS_STREAM_NAME_DEFAULT),
+            Importance.MEDIUM,
+            "The name of the migration stream topic. It defaults to "
+        ).define(
+            KSQL_MIGRATIONS_TABLE_TOPIC_NAME,
+            Type.STRING,
+            id + "ksql_" + configs.getOrDefault(KSQL_MIGRATIONS_TABLE_NAME, KSQL_MIGRATIONS_TABLE_NAME_DEFAULT),
+            Importance.MEDIUM,
+            "The name of the migration table topic. It defaults to"
+        ).define(
+            KSQL_MIGRATIONS_TOPIC_REPLICAS,
+            Type.INT,
+            KSQL_MIGRATIONS_TOPIC_REPLICAS_DEFAULT,
+            Importance.MEDIUM,
+            "The number of replicas for the migration stream topic. It defaults to " +
+                KSQL_MIGRATIONS_TOPIC_REPLICAS_DEFAULT
+        ), configs);
+  }
+
+  private static String getServiceId(final Map<String, String> configs) {
+    final String ksqlServerUrl = configs.get(MigrationConfig.KSQL_SERVER_URL);
+    if (ksqlServerUrl == null) {
+      LOGGER.error("Missing required property: " + MigrationConfig.KSQL_SERVER_URL);
+      System.exit(1);
+    }
+
+    final KsqlRestClient client;
+    try {
+      client = KsqlRestClient.create(
+          ksqlServerUrl,
+          Collections.EMPTY_MAP,
+          Collections.EMPTY_MAP,
+          Optional.empty()
+      );
+    } catch (KsqlRestClientException e) {
+      LOGGER.error("Invalid ksql server URL: " + ksqlServerUrl);
+      System.exit(1);
+      return null;
+    }
+
+    final RestResponse<ServerInfo> response = client.getServerInfo();
+    client.close();
+
+    if (!response.isSuccessful()) {
+      LOGGER.error("Failed to query " + MigrationConfig.KSQL_SERVER_URL + "/info");
```

Comment:
Don't worry about this for now since this code is going to be replaced with Java client code anyway, but after the replacement we should verify that this error message is actionable. For example, the current error message doesn't give any information regarding why the failure occurred (auth issue, transient network issue, invalid address, etc).